keyboards/system76/launch_1/keymaps/default/keymap.c

```diff
@@ -87,5 +90,22 @@ ________________________________________________________________________________
 };
 
 bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+    case KC_LCTL:
+        lctl_pressed = record->event.pressed;
+        break;
+    case KC_RCTL:
+        rctl_pressed = record->event.pressed;
+        break;
+    case KC_ESC:
+        esc_pressed = record ->event.pressed;
+        break;
+    };
     return true;
 }
+
+void matrix_scan_user(void) {
```

Comment:
If we want this to be universal (all keymaps) I believe we could do process_record_keyboard and matrix_scan_keyboard, what do you think?

Reply:
I like that idea. Between this and your other comment, I've come up with a new approach: https://github.com/system76/qmk_firmware/pull/24/commits/a779f0c93cc738f25a5e16b87b1fc64e796616a7